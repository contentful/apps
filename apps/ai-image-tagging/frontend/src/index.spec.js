import React from 'react';
import { App } from './index';
import { render, cleanup, configure } from '@testing-library/react';

configure({
  testIdAttribute: 'data-test-id'
});

function renderComponent(sdk) {
  return render(<App sdk={sdk} />);
}

const sdk = {
  entry: {
    fields: {
      image: {
        getValue: jest.fn(),
        onValueChanged: jest.fn()
      },
      imageTags: {
        getValue: jest.fn(),
        setValue: jest.fn()
      }
    }
  },
  notifier: {
    error: jest.fn()
  },
  window: {
    startAutoResizer: jest.fn(),
  },
  app: {
    setReady: jest.fn()
  },
  locales: {
    default: 'en-US'
  },
  space: {
    getAsset: jest.fn()
  }
};

describe('App', () => {
  beforeEach(jest.resetAllMocks);

  afterEach(cleanup);

  it('should call starstartAutoResizer', () => {
    renderComponent(sdk);
    expect(sdk.window.startAutoResizer).toHaveBeenCalled();
  });

  it('should call sdk ready fn', () => {
    renderComponent(sdk);
    expect(sdk.app.setReady).toHaveBeenCalled();
  });

  describe('#render', () => {
    it('should render the extension field view', () => {
      const { container } = renderComponent(sdk);
      expect(container).toMatchSnapshot();
    });
  });
});
