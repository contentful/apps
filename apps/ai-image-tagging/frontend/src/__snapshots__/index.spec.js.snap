// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App #render should render the extension field view 1`] = `
<div>
  <div>
    <div
      class="TextInput__TextInput___36-K- TextInput__TextInput--large___KwY4O TextInput__TextInput--disabled___2t7VS"
    >
      <input
        class="TextInput__TextInput__input___27vDB a11y__focus-border--default___60AXp"
        data-test-id="image-tag"
        disabled=""
        placeholder="Type a tag and press enter"
        value=""
      />
    </div>
    <div
      class="css-m9mlq4"
    />
    <button
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-1kr4fmq Button__Button--primary___JImeO Button__Button--disabled___1E20M"
      data-test-id="cf-ui-button"
      disabled=""
      id="fetch-tag-btn"
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          Auto-tag from AI
        </span>
      </span>
    </button>
    <div
      class="ControlledInputField__ControlledInputField___2uIG9 ControlledInputField__ControlledInputField--disabled___HDWt6"
      data-test-id="cf-ui-checkbox-field"
    >
      <input
        aria-label="Overwrite existing tags"
        checked=""
        class="ControlledInput__ControlledInput___2XK3j ControlledInputField__ControlledInputField__input___3OMYB ControlledInput__ControlledInput--disabled___3prPF"
        data-test-id="cf-ui-controlled-input"
        disabled=""
        id="overwrite-tags"
        type="checkbox"
        value=""
      />
      <div>
        <label
          class="FormLabel__FormLabel___3d6zQ ControlledInputField__ControlledInputField__label___a9J52"
          data-test-id="cf-ui-form-label"
          for="overwrite-tags"
        >
          Overwrite existing tags
        </label>
      </div>
    </div>
  </div>
</div>
`;
